### ISSUE TITLE
Add monitoring options to aws_glue_job resource

### ISSUE THREAD

**Issue by kabeersvohra** [terraform-providers/terraform-provider-aws#10946]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Can we please add support for monitoring options to the `aws_glue_job` resource? Specifically `Job metrics`, `Continuous logging` and `Spark UI` options.

### New or Affected Resource(s)

* aws_glue_job

### Potential Terraform Configuration

```hcl
resource "aws_glue_job" "job_id" {
  name = "job-name"
  monitoring {
    job_metrics = true
    continuous_logging = true
    spark_ui = false
  }
}
```

**Comment by Zambonilli** [comment]
spark ui can be set via the following default-arguments:
 "--enable-spark-ui"       = "true"
 "--spark-event-logs-path" = "s3://${my-s3-bucket}/glue/sparkui/"

**Comment by bflad** [comment]
Hi @kabeersvohra 👋 Thank you for submitting this feature request.

The [Glue Jobs API](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-jobs-job.html) does not include specific fields for configuring these options. Similar to the previous comment, many of these options seem to be part of submitting [special parameters](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html) as arguments to the Glue Job. We typically avoid introducing our own abstractions into the Terraform AWS Provider as they become easily out of date and a maintenance burden.

To enable CloudWatch Logs and metrics with the current resource:

```hcl
resource "aws_cloudwatch_log_group" "example" {
  name              = "example"
  retention_in_days = 14
}

resource "aws_glue_job" "example" {
  # ... other configuration ...

  default_arguments = {
    # ... potentially other arguments ...
    "--continuous-log-logGroup"          = "${aws_cloudwatch_log_group.example.name}"
    "--enable-continuous-cloudwatch-log" = "true"
    "--enable-continuous-log-filter"     = "true"
    "--enable-metrics"                   = ""
  }
}
```

I have submitted an update to the resource documentation to include additional information about these topics: https://github.com/terraform-providers/terraform-provider-aws/pull/11009

**Comment by bflad** [comment]
The documentation update above has been merged and will publish when version 2.40.0 is published later this week. 👍

**Comment by kabeersvohra** [comment]
Hi @bflad. Sorry for the late reply! Yes I see that now :), thank you for adding it into the documentation!
